src/sparc/client/services/metadata.py

```diff
@@ -0,0 +1,278 @@
+            logging.info("Profile: none")
+        if connect:
+            self.connect()
+
+    def connect(self) -> str:
+        """Not needed as metadata services are REST service calls"""
+        logging.info("Metadata REST services available...")
+
+        self.host_api = "https://scicrunch.org/api/1/elastic"
+        return self.host_api
+
+    def info(self) -> str:
+        """Returns information about the metadata search services."""
+
+        self.host_api = "https://scicrunch.org/api/1/elastic"
+        return self.host_api
+
+    def get_profile(self) -> str:
+        """Returns currently used API key.
+
+        Returns:
+        --------
+        A string with API Key.
+        """
+        return self.scicrunch_api_key
+
+    def set_profile(self, api_key: str) -> str:
+        """Changes the API key to the specified name.
+
+        Parameters:
+        -----------
+        api_key : str
+            The API key to use.
+
+        Returns:
+        --------
+        A string with confirmation of API key switch.
+        """
+        self.scicrunch_api_key = api_key
+        return self.scicrunch_api_key
+
+    def close(self) -> None:
+        """Not needed as metadata services are REST service calls"""
+        return self.host_api
+
+    #####################################################################
+    # Supporting Functions
+
+    #####################################################################
+    # Function to GET content from URL with retries
+    def getURL(self, url, headers="NONE"):
+        result = "[ERROR]"
+        url_session = requests.Session()
+
+        retries = Retry(
+            total=6, backoff_factor=1, status_forcelist=[403, 404, 413, 429, 500, 502, 503, 504]
+        )
+
+        url_session.mount("https://", HTTPAdapter(max_retries=retries))
+
+        success = 1
+
+        try:
+            if headers == "NONE":
+                url_result = url_session.get(url)
+            else:
+                url_result = url_session.get(url, headers=headers)
+
+            if url_result.status_code == 410:
+                logging.warning("Retrieval Status 410 - URL Unpublished:" + url)
+            else:
+                url_result.raise_for_status()
+
+        except requests.exceptions.HTTPError as errh:
+            logging.error("Retrieving URL - HTTP Error:", errh)
+            success = 0
+        except requests.exceptions.ConnectionError as errc:
+            logging.error("Retrieving URL - Error Connecting:", errc)
+            success = 0
+        except requests.exceptions.Timeout as errt:
+            logging.error("Retrieving URL - Timeout Error:", errt)
+            success = 0
+        except requests.exceptions.RequestException as err:
+            logging.error("Retrieving URL - Something Else", err)
+            success = 0
+
+        url_session.close()
+
+        if success == 1:
+            result = url_result
+        else:
+            result = {}
```

Comment:
I am a bit confused what `result` is doing. 
First we set it as "[ERROR]" then overwriting with url_results. 
I believe all 3 variables: `success`, `result` and `url_results` could be combined into a single variable.

Reply:
Should be done now.